pkg/image/api/v1/trigger/trigger.go

```diff
@@ -0,0 +1,24 @@
+package trigger
+
+// ObjectFieldTrigger links a field on the current object to another object for mutation.
+type ObjectFieldTrigger struct {
+	// from is the object this should trigger from. The kind and name fields must be set.
+	From ObjectReference `json:"from"`
+	// fieldPath is a JSONPath string to the field to edit on the object. Required.
+	FieldPath string `json:"fieldPath"`
+	// paused is true if this trigger is temporarily disabled. Optional.
+	Paused bool `json:"paused,omitempty"`
+}
+
+// ObjectReference identifies an object by its name and kind.
+type ObjectReference struct {
```

Comment:
So that it's not dependent on kapi.  Also doesn't need the same fields.  Still effectively versioned, but not tied to Kube.